website: dependency bumps and handle multi-exports

rjspiker opened the pull request:
This bumps versions to the latest components.  A couple of components now have multiple exports, so this also makes sure the default export is being used for those.

cc: @jescalan

rjspiker commented on website/assets/package.json:
```diff
@@ -10,22 +10,22 @@
-    "@hashicorp/hashi-image": "1.0.4",
+    "@hashicorp/hashi-global-styles": "^1.0.9",
+    "@hashicorp/hashi-hero": "^2.1.9",
+    "@hashicorp/hashi-image": "1.0.5",
```
However, we're never actually exporting image ourselves.  It's simply there because it's a peer dependency being used by hero and downloads (I think it's just those two).  We don't export it and use it anywhere ourselves.

Please tell me if I'm misunderstanding something here.

jescalan replied:
Ah in that case we don't need to add it to reshape then right? That only rehydrates top level elements

rjspiker replied:
Right.  It's just a peer dep.  We aren't exporting in `reshape.js`.

rjspiker commented on website/assets/package.json:
```diff
@@ -10,22 +10,22 @@
+    "@hashicorp/hashi-image": "1.0.5",
     "@hashicorp/hashi-linked-text-summary-list": "^0.1.4",
-    "@hashicorp/hashi-logo-grid": "2.2.1",
+    "@hashicorp/hashi-logo-grid": "2.2.2",
```
This is the same scenario as hashi-image from what I'm seeing, except I'm not sure that we actually use this anywhere at all or if anything is depending on this at all.  We are importing it's styles into the `index.css`, but I can't see the component being used anywhere.  I don't want to just remove it, though.  Do you know if it's actually being used at all?  Am I overlooking a something?

Again, please tell me if I'm misunderstanding something here.

jescalan replied:
It definitely could not be used - this template was based off of the sentinel docs so it's probably just a relic. If you don't see it anywhere in the source feel free to remove!

rjspiker commented on website/assets/package.json:
```diff
@@ -10,22 +10,22 @@
-    "@hashicorp/hashi-newsletter-signup-form": "1.0.6",
-    "@hashicorp/hashi-product-downloader": "^0.5.4",
+    "@hashicorp/hashi-newsletter-signup-form": "1.1.1",
+    "@hashicorp/hashi-product-downloader": "^0.6.0",
```
Thanks.  Rebased.

rjspiker commented:
@jescalan OK, I think we're good now.  Rebased to get the updated `hashi-product-downloader`.  Removed the dependency on `hashi-logo-grid`.  Left the `hashi-image` as is.
